Error when building with boost-1.89.0

### Performous & operating system versions

Performous: latest master branch
Operating System: Arch Linux

### What is the problem?

When trying to build on a current Arch Linux, the build fails.

### Steps to reproduce

1. clone the repo
2. follow wiki instructions on building (it fails after cmake ..)

### Additional context

Output of cmake .. :

```
No project version specified. Determining latest tag..
Building Performous 1.3.1-281-g94f237f8
-- Localization enabled: Building and installing .mo files
-- Using legacy Boost variables
-- Found LibEpoxy 1.5.10
-- SDL2 includes: /usr/include;/usr/include/SDL2
-- Found PangoCairo 
-- PangoCairo includes: 
-- Found Freetype 
-- Found Cairo 1.18.4
-- Found GDK-PixBuf 
-- Found LibRSVG 2.61.1
-- LibRSVG includes: /usr/include/librsvg-2.0;/usr/include/cairo;/usr/include/freetype2;/usr/include/gdk-pixbuf-2.0
-- Found LibXML2 
-- Found SigC++ 3.0
-- Found Glibmm 2.68
-- Found LibXML++ 2.6
-- LibXML++ includes: /usr/lib/libxml++-2.6/include;/usr/include/libxml++-2.6;/usr/include/libxml2;/usr/lib/glibmm-2.68/include;/usr/include/glibmm-2.68;/usr/include/sigc++-3.0;/usr/lib/sigc++-3.0/include
-- Found AVUtil 
-- Found AVCodec 
-- Found AVFormat 
-- AVFormat includes: /usr/include;/usr/include;/usr/include
-- Found SWResample 
-- SWResample includes: /usr/include;/usr/include
-- Found SWScale 
-- SWScale includes: /usr/include;/usr/include
-- ZLIB includes: /usr/include
-- JPEG includes: /usr/include
-- PNG includes: /usr/include;/usr/include
-- Found WebP 1.6.0
-- WebP includes: /usr/include
-- Found PortAudio 19
-- PortAudio includes: /usr/include
-- Found Fontconfig 
-- Fontconfig includes: /usr/include
-- GLM_INCLUDE_DIR = /usr/include
-- GLM includes: /usr/include
-- Found Json 
-- Json includes: /home/rex/performous/build/game/json-src-src/include
-- Checking for module 'CED'
--   Package 'CED' not found
-- ced build from source because not found on system
-- Fetching and making available ced-src...
-- Found Google CED master
-- Ced includes: /home/rex/performous/build/game/ced-src-src
-- Found Aubio 0.4.9
-- Aubio includes: /usr/include
-- FMT VERSION: 12.1.0
-- Found Spdlog 
-- MIDI I/O support: Enabled (automatically found)
-- Webcam support: Enabled (automatically found)
-- Webserver support: Enabled (automatically found)
-- Testing enabled: Building test host
CMake Error at /usr/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_system"
  (requested version 1.89.0) with any of the following names:

    boost_systemConfig.cmake
    boost_system-config.cmake

  Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
  "boost_system_DIR" to a directory containing one of the above files.  If
  "boost_system" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /usr/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
  /usr/share/cmake/Modules/FindBoost.cmake:610 (find_package)
  testing/CMakeLists.txt:53 (find_package)


-- Configuring incomplete, errors occurred!

```

### Log(s) & preferences file(s)

_No response_

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -107,12 +107,6 @@ jobs:
           - os: debian
             version: 13
           - os: fedora
-            version: 36
-          - os: fedora
-            version: 37
-          - os: fedora
-            version: 38
-          - os: fedora
             version: 39
           - os: fedora
             version: 40

--- a/game/CMakeLists.txt
+++ b/game/CMakeLists.txt
@@ -85,14 +85,8 @@ set(SOURCES ${SOURCE_FILES} ${HEADER_FILES} ${RESOURCE_FILES})
 add_executable(performous ${SUBSYSTEM_WIN32} ${SOURCES})
 # Libraries
 
-find_package(Boost 1.55 CONFIG QUIET COMPONENTS program_options iostreams locale OPTIONAL_COMPONENTS system)
-if(NOT Boost_FOUND)
-  find_package(Boost 1.55 REQUIRED COMPONENTS program_options iostreams locale OPTIONAL_COMPONENTS system)
-endif()
-if(Boost_VERSION VERSION_LESS "1.69.0" AND NOT TARGET Boost::system)
-  message(FATAL_ERROR "Boost::system component is required but not found (Boost ${Boost_VERSION})")
-endif()
-target_link_libraries(performous PRIVATE Boost::boost Boost::program_options Boost::iostreams Boost::locale $<TARGET_NAME_IF_EXISTS:Boost::system>)
+find_package(Boost 1.70 CONFIG REQUIRED COMPONENTS program_options iostreams locale)
+target_link_libraries(performous PRIVATE Boost::boost Boost::program_options Boost::iostreams Boost::locale)
 
 find_package(ICU 65 REQUIRED uc data i18n io)
 target_link_libraries(performous PRIVATE ICU::uc ICU::data ICU::i18n ICU::io)
@@ -224,7 +218,7 @@ endif()
 if(cpprestsdk_FOUND OR CppRest_FOUND)
 	# add other required dependencies then
 	# FIXME: move this to the CppRest package includes if possible
-	find_package(Boost 1.55 REQUIRED COMPONENTS chrono thread)
+	find_package(Boost 1.70 CONFIG REQUIRED COMPONENTS chrono thread)
 	find_package(OpenSSL REQUIRED)
 	if(cpprestsdk_FOUND)
 		target_link_libraries(performous PRIVATE cpprestsdk::cpprest)

--- a/testing/CMakeLists.txt
+++ b/testing/CMakeLists.txt
@@ -50,11 +50,8 @@ if(GTest_FOUND)
 	
 	add_executable(performous_test ${SOURCES})
 
-	find_package(Boost 1.55 REQUIRED COMPONENTS program_options iostreams locale OPTIONAL_COMPONENTS system)
-	if(Boost_VERSION VERSION_LESS "1.69.0" AND NOT TARGET Boost::system)
-		message(FATAL_ERROR "Boost::system component is required but not found (Boost ${Boost_VERSION})")
-	endif()
-	target_link_libraries(performous_test PRIVATE Boost::boost Boost::iostreams Boost::program_options Boost::locale $<TARGET_NAME_IF_EXISTS:Boost::system>)
+	find_package(Boost 1.70 CONFIG REQUIRED COMPONENTS program_options iostreams locale)
+	target_link_libraries(performous_test PRIVATE Boost::boost Boost::iostreams Boost::program_options Boost::locale)
 
 	find_package(fmt 8.1 REQUIRED CONFIG)
 	target_link_libraries(performous_test PRIVATE fmt::fmt)

